src/native/fs/fs_napi.cpp

```diff
@@ -991,6 +995,38 @@ struct FileSetxattr : public FSWrapWorker<FileWrap>
     }
 };
 
+struct linkFileAt : public FSWrapWorker<FileWrap>
+{
+    std::string _path;
+    std::string _filepath;
+    int _dirfd;
+    int _rc_res;
+
+
+    linkFileAt(const Napi::CallbackInfo& info)
+        : FSWrapWorker<FileWrap>(info)
+    {
+        _path = info[1].As<Napi::String>();
+        _filepath = info[2].As<Napi::String>();
+        _dirfd = _wrap->_fd;
```

Comment:
wait, this is not a dir here, we are implementing this for FileWrap, meaning that _wrap->_fd is of the file that we opened, which is the temp file itself. So is it just an incorrect name of the variable? or did you expect to find the open directory fd here?

Reply:
also there is no need for you to copy it, you can just access _wrap->_fd inside Work()...

Reply:
You add at the end of the ctor this call like all other workers do:
```
        Begin(XSTR() << "LinkFileAt " << DVAL(_wrap->_path) << DVAL(_wrap->_fd) << DVAL(_filepath));
```

Reply:
it is the file fd, I removed _dirfd